doc/modules/grid_search.rst

```diff
@@ -192,11 +192,46 @@ result in an error when using multiple metrics.
 See :ref:`sphx_glr_auto_examples_model_selection_plot_multi_metric_evaluation.py`
 for an example usage.
 
+.. _composite_grid_search:
+
 Composite estimators and parameter spaces
 -----------------------------------------
+`GridSearchCV` and `RandomizedSearchCV` allow searching over parameters of
+composite or nested estimators such as `Pipeline`, `ColumnTransformer`,
+`VotingClasssifier` or `CalibratedClassifierCV`
+using a dedicated syntax ``<estimator>__<parameter>`` syntax::
+
+  >>> from sklearn.model_selection import GridSearchCV
+  >>> from sklearn.calibration import CalibratedClassifierCV
+  >>> from sklearn.ensemble import RandomForestClassifier
+  >>> from sklearn.datasets import make_moons
+  >>> X, y = make_moons()
+  >>> calibrated_forest = CalibratedClassifierCV(
+  ...    base_estimator=RandomForestClassifier(n_estimators=10))
+  >>> param_grid = {
+  ...    'base_estimator__max_depth': [2, 4, 6, 8]}
+  >>> search = GridSearchCV(calibrated_forest, param_grid, cv=5)
+  >>> search.fit(X, y)
+  GridSearchCV(cv=5,
+               estimator=CalibratedClassifierCV(base_estimator=RandomForestClassifier(n_estimators=10)),
+               param_grid={'base_estimator__max_depth': [2, 4, 6, 8]})
```

Comment:
I was thinking of using ellipsis, which passes pytest.

```py
  GridSearchCV(cv=5,
               estimator=CalibratedClassifierCV(...),
               param_grid={'base_estimator__max_depth': [2, 4, 6, 8]})
```

Depends on if we prefer this:

![Screen Shot 2019-08-05 at 4 54 51 PM](https://user-images.githubusercontent.com/5402633/62494560-d3a60000-b7a1-11e9-94ec-d557e48de1a9.png)

or this:

![Screen Shot 2019-08-05 at 4 55 40 PM](https://user-images.githubusercontent.com/5402633/62494582-e0c2ef00-b7a1-11e9-9f8a-d4ed2fb6955c.png)